docs/docker-compose/examples/with-telegraf-metrics.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Collecting Alfresco metrics with Telegraf
+parent: Docker Compose
+---
+
+# Collecting Alfresco metrics with Telegraf
+
+This guide aims to demonstrate how to set up Telegraf to collect JMX and
+Micrometer metrics from an Alfresco Content Services instance.
+
+## Prerequisites
+
+This example uses values files from `docs/docker-compose/examples/values` folder:
+
+- telegraf-overrides.yaml
+- telegraf.conf
+
+To ease the setup process, you can copy these files to to `docker-compose` folder.
+
+For Jolokia to work it is required to download the
+[Jolokia JVM agent jar](https://search.maven.org/remotecontent?filepath=org/jolokia/jolokia-agent-jvm/2.3.0/jolokia-agent-jvm-2.3.0-javaagent.jar) file and place it in `docker-compose` folder.
+
+## Running the Example
+
+When the files are in place, you can start the ACS instance with Telegraf enabled:
+
+```bash
+docker-compose -f compose.yaml -f telegraf-overrides.yaml up -d
+```
+
+This will start the ACS instance with Telegraf configured to collect Jolokia and
+Micrometer metrics and ship them to an InfluxDB instance.
+
+## Micrometer Configuration
+
+To enable Micrometer metrics in Alfresco Content Services, the following Java
+system properties need to be set:
+
+```yaml
+    environment:
+      JAVA_OPTS: >-
+        -Dmetrics.enabled=true
+        -Dmetrics.dbMetricsReporter.enabled=true
+        -Dmetrics.dbMetricsReporter.query.enabled=true
+        -Dmetrics.dbMetricsReporter.query.statements.enabled=true
+        -Dmetrics.jvmMetricsReporter.enabled=true
+        -Dmetrics.restMetricsReporter.enabled=true
+        -Dmetrics.restMetricsReporter.path.enabled=true
+        -Dmetrics.tomcatMetricsReporter.enabled=true
+        -Dmetrics.authenticationMetricsReporter.enabled=true
+```
+
+These properties are already included in the `telegraf-overrides.yaml` file.
+This configuration enables repository to expose metrics on
+`http://localhost:8080/alfresco/s/prometheus` endpoint.
+
+Telegraf is configured to collect these metrics with the following section in
+the `telegraf.conf` file:
+
+```toml
+[[inputs.prometheus]]
+  urls = [
+    "http://alfresco:8080/alfresco/s/prometheus"
+  ]
+```
+
+The hostname `alfresco` is used as the Telegraf container and Alfresco Content
+Services container are part of the same Docker network created by Docker
+Compose. This configuration is specific to the
+[prometheus input plugin](https://github.com/influxdata/telegraf/tree/release-1.35/plugins/inputs/prometheus).
+
+Shipping Micrometer metrics to InfluxDB is configured with the following section
+in the `telegraf.conf` file:
+
+```toml
+[[outputs.influxdb_v2]]
+  bucket = "alfresco"
+  organization = "alfresco"
+  timeout = "5s"
+  token = "influx"
+  urls = [
+    "http://influxdb2:8086"
+  ]
+```
```

Comment:
not sure if make sense to document influx specifically here, but anyway telegraf is part of the same ecosystem so it's a natural choice. we should probably specify better in the introduction that we are targeting influxdb but we can target any supported output (link to the outputs?)